### PR TITLE
￼ fix:allow customer and supplier codes bigger than 15 chars

### PR DESCRIPTION
**Pull request by bafbes**
# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "Fix", "Close" or "New" section*
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed textswith meaningful informations*

# Fix #[*issue_number Short description*]
[*Long description*]

# Close #[*issue_number Short description*]
[*Long description*]

# New [*Short description*]
[*Long description*]

### CODE REVIEW

File: htdocs/societe/card.php [hunk 1]
```diff
@@ -2224,6 +2225,344 @@ function init_supplier_categ() {
 				print $form->select_incoterms((!empty($object->fk_incoterms) ? $object->fk_incoterms : ''), (!empty($object->location_incoterms) ? $object->location_incoterms : ''));
 				print '</td></tr>';
 			}
+=======
```
Comment (eldy) [inline htdocs/societe/card.php:2228]: What is this ?

File: htdocs/societe/card.php [hunk 2]
```diff
@@ -1186,493 +1186,493 @@ function init_check_no_email(input) {
...
-
-			print '<td>';
-			if (!empty($conf->fournisseur->enabled) && !empty($user->rights->fournisseur->lire))
-			{
-				print $form->editfieldkey('SupplierCode', 'supplier_code', '', $object, 0);
-			}
+			print '<table class="nobordernopadding"><tr><td>';
+			$tmpcode = $object->code_fournisseur;
+			if (empty($tmpcode) && !empty($modCodeFournisseur->code_auto)) $tmpcode = $modCodeFournisseur->getNextValue($object, 1);
+			print '<input type="text" name="supplier_code" id="supplier_code" class="maxwidthonsmartphone" value="'.dol_escape_htmltag($tmpcode).'" >';
```
Comment (eldy) [inline htdocs/societe/card.php:1324]: What happen if we enter a value with 300 chars ?

**Comment by bafbes** [comment]
I think that only 24 chars are saved as limited in llx_societe table with code_client varchar(24)

**Comment by eldy** [comment]
With another PR, i modified max from 15 to 24

**Pull request closed without merging**